### ISSUE TITLE
Keras implementation and model structure

### ISSUE THREAD

**Issue by shekarneo** [sushant097/Handwritten-Line-Text-Recognition-using-Deep-Learning-with-Tensorflow#7]
Hello,

i am trying to implement this with keras,
in your code (Model.py) you have mentioned the layer filters, but i am not sure about the rnn and ctc layers.

please have a look at this

![model](https://user-images.githubusercontent.com/25033310/73261461-087c9800-41c4-11ea-89de-707edb827bda.png)

**Comment by shekarneo** [comment]
> Hello,
> 
> i am trying to implement this with keras,
> in your code (Model.py) you have mentioned the layer filters, but i am not sure about the rnn and ctc layers.
> 
> please have a look at this
> 
> # `_________________________________________________________________
> Layer (type) Output Shape Param #
> the_input (InputLayer) (None, 800, 64, 1) 0
> 
> conv1 (Conv2D) (None, 800, 64, 64) 1664
> 
> pool1 (MaxPooling2D) (None, 400, 32, 64) 0
> 
> conv2 (Conv2D) (None, 400, 32, 128) 204928
> 
> pool2 (MaxPooling2D) (None, 400, 16, 128) 0
> 
> conv3 (Conv2D) (None, 400, 16, 128) 147584
> 
> pool3 (MaxPooling2D) (None, 200, 8, 128) 0
> 
> BN1 (BatchNormalization) (None, 200, 8, 128) 512
> 
> conv4 (Conv2D) (None, 200, 8, 256) 295168
> 
> conv5 (Conv2D) (None, 200, 8, 256) 590080
> 
> pool4 (MaxPooling2D) (None, 100, 4, 256) 0
> 
> conv6 (Conv2D) (None, 100, 4, 512) 1180160
> 
> pool5 (MaxPooling2D) (None, 100, 2, 512) 0
> 
> BN2 (BatchNormalization) (None, 100, 2, 512) 2048
> 
> conv7 (Conv2D) (None, 100, 2, 512) 2359808
> 
> pool6 (MaxPooling2D) (None, 100, 1, 512) 0
> 
> reshape (Reshape) (None, 100, 512) 0
> 
> bidirectional_8 (Bidirection (None, 1024) 4198400
> 
> # Dense1 (Dense) (None, 100) 102500
> Total params: 9,082,852
> Trainable params: 9,081,572
> Non-trainable params: 1,280`

**Comment by sushant097** [comment]
It seems to be everything right! how about CTC layers on keras? I have no idea about ctc_loss in keras which seems to have some error in merging with this work.

**Comment by shekarneo** [comment]
![model1](https://user-images.githubusercontent.com/25033310/74003305-09eb5480-496a-11ea-8e1e-e2c1d2d5e526.png)
have created the fully model with CTC but i am getting the sequence length error

**Comment by sushant097** [comment]
I have no idea about this. Maybe this [Link](https://theailearner.com/2019/05/29/creating-a-crnn-model-to-recognize-text-in-an-image-part-2/) can be helpful for you. Keras provides the wrapper for the CTC loss of TensorFlow. Maybe TensorFlow 2.0 implementation of CRNN can be helpful.
